OSDOCS-13469-19: Release notes file template (#89357)
openshift/openshift-docs: release_notes/ocp-4-19-release-notes.adoc

|Supported
|Supported

|OpenShift Elasticsearch Operator
|Supported
|Supported

|Vertical Pod Autoscaler Operator
|Supported
|Supported
|====

.Multus CNI plugins
[cols="2,1,1",options="header"]
|====
|Feature |{ibm-power-name} |{ibm-z-name} and {ibm-linuxone-name}

|Bridge
|Supported
|Supported

|Host-device
|Supported
|Supported

|IPAM
|Supported
|Supported

|IPVLAN
|Supported
|Supported
|====

.CSI Volumes
[cols="2,1,1",options="header"]
|====
|Feature |{ibm-power-name} |{ibm-z-name} and {ibm-linuxone-name}

|Cloning
|Supported
|Supported

|Expansion
|Supported
|Supported

|Snapshot
|Supported
|Supported
|====

[id="ocp-release-notes-insights-operator-enhancements_{context}"]
=== Insights Operator

[id="ocp-release-notes-installation-and-update_{context}"]
=== Installation and update

[id="ocp-release-notes-machine-config-operator_{context}"]
=== Machine Config Operator

[id="ocp-release-notes-management-console_{context}"]
=== Management console

[id="ocp-release-notes-monitoring_{context}"]
=== Monitoring

[id="ocp-release-notes-network-observability-operator_{context}"]
=== Network Observability Operator

[id="ocp-release-notes-networking_{context}"]
=== Networking

> 

Reviewer: ```suggestion

[id="ocp-4-19-networking-gateway-api_{context}"]
==== Support for using the Gateway API and Istio to manage cluster Ingress operations
We we have details here when OpenShift Container Platform 4.19 is released

```
@Miciah what do you think?

Reviewer 2: Thanks!  Having this placeholder for a referent will be very helpful in unblocking our 4.18 pre-upgrade check.  

Two suggestions:

- First, we decided to split the Gateway API work into two separate featuregates, and it makes sense to call both of these out, as separate features:
  1. _Gateway API CRD life-cycle management_, meaning that the platform (that is, OCP itself) manages the Gateway API CRDs.  
  2. The _Gateway API controller_, meaning that the platform also has a controller that makes useful things happen when you create resources using these CRDs.
- Second, we want to de-emphasize the fact that we are using OSSM and Istio/Envoy underneath.  That fact won't be entirely hidden, but it shouldn't be part of the branding, and I think it would be better to omit "Istio" (or Envoy, or "OSSM", "Service Mesh", etc.) from the heading.  So something like "Support for using the Gateway API to configure cluster Ingress traffic" would work.

|Technology Preview
|Technology Preview
|Technology Preview

|VRF specification in `NodeNetworkConfigurationPolicy` custom resource
|Technology Preview
|Technology Preview
|Technology Preview

|Host network settings for SR-IOV VFs
|General Availability
|General Availability
|General Availability

|Integration of MetalLB and FRR-K8s
|General Availability
|General Availability
|General Availability

|Automatic leap seconds handling for PTP grandmaster clocks
|General Availability
|General Availability
|General Availability

|PTP events REST API v2
|General Availability
|General Availability
|General Availability

|Customized `br-ex` bridge needed by OVN-Kuberenetes to use NMState
|Technology Preview
|General Availability
|General Availability

|Live migration to OVN-Kubernetes from OpenShift SDN
|General Availability
|Not Available
|Not Available

|User-defined network segmentation
|Technology Preview
|Technology Preview
|General Availablity

|Dynamic configuration manager
|Not Available
|Technology Preview
|Technology Preview

|SR-IOV Network Operator support for Intel C741 Emmitsburg Chipset
|Not Available
|Technology Preview
|Technology Preview

> |====

Reviewer: ```suggestion

```

> 

Reviewer: ```suggestion

|Gateway API and Istio for Ingress management
| Not Available
| Not Available
| General Availability
```
@Miciah PTAL

|Supported
|Supported

|OpenShift Elasticsearch Operator
|Supported
|Supported

|Vertical Pod Autoscaler Operator
|Supported
|Supported
|====

.Multus CNI plugins
[cols="2,1,1",options="header"]
|====
|Feature |{ibm-power-name} |{ibm-z-name} and {ibm-linuxone-name}

|Bridge
|Supported
|Supported

|Host-device
|Supported
|Supported

|IPAM
|Supported
|Supported

|IPVLAN
|Supported
|Supported
|====

.CSI Volumes
[cols="2,1,1",options="header"]
|====
|Feature |{ibm-power-name} |{ibm-z-name} and {ibm-linuxone-name}

|Cloning
|Supported
|Supported

|Expansion
|Supported
|Supported

|Snapshot
|Supported
|Supported
|====

[id="ocp-release-notes-insights-operator-enhancements_{context}"]
=== Insights Operator

[id="ocp-release-notes-installation-and-update_{context}"]
=== Installation and update

[id="ocp-release-notes-machine-config-operator_{context}"]
=== Machine Config Operator

[id="ocp-release-notes-management-console_{context}"]
=== Management console

[id="ocp-release-notes-monitoring_{context}"]
=== Monitoring

[id="ocp-release-notes-network-observability-operator_{context}"]
=== Network Observability Operator

[id="ocp-release-notes-networking_{context}"]
=== Networking

> [id="ocp-4-19-networking-gateway-api_{context}"]
> ==== Support for using the Gateway API and Istio to manage cluster Ingress operations
> //We will have details here when {product-title} {product-version} is released.

Reviewer: ```suggestion
[id="ocp-4-19-networking-gateway-api-controller_{context}"]
==== Support for using the Gateway API to configure cluster Ingress traffic
We will have details here when {product-title} {product-version} is released.

[id="ocp-4-19-networking-gateway-api-crd-lifecycle_{context}"]
==== Support for managing the Gateway API custom resource definition (CRD) lifecycle
We will have details here when {product-title} {product-version} is released.
```

//Odd-numbered release lifecycle verbiage (Comment in for odd-numbered releases)
////
The support lifecycle for odd-numbered releases, such as {product-title} {product-version}, on all supported architectures, including `x86_64`, 64-bit ARM (`aarch64`), {ibm-power-name} (`ppc64le`), and {ibm-z-name} (`s390x`) architectures is 18 months. For more information about support for all versions, see the link:https://access.redhat.com/support/policy/updates/openshift[Red Hat {product-title} Life Cycle Policy].
////

> Commencing with the {product-title} 4.14 release, Red{nbsp}Hat is simplifying the administration and management of Red{nbsp}Hat shipped cluster Operators with the introduction of three new life cycle classifications; Platform Aligned, Platform Agnostic, and Rolling Stream. These life cycle classifications provide additional ease and transparency for cluster administrators to understand the life cycle policies of each Operator and form cluster maintenance and upgrade plans with predictable support boundaries. For more information, see link:https://access.redhat.com/webassets/avalon/j/includes/session/scribe/?redirectTo=https%3A%2F%2Faccess.redhat.com%2Fsupport%2Fpolicy%2Fupdates%2Fopenshift_operators[OpenShift Operator Life Cycles].

Reviewer 3: Out of scope: I wonder if/when we ever stop carrying over the "from 4.14" (or 4.16) language in the RNs?

Author: These statements are complex but I think when 4.14 enters sustainability engineers purview.

=== Images deprecated and removed features

.Images deprecated and removed tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.17 |4.18 |4.19

|Cluster Samples Operator
|Deprecated
|Deprecated
|Deprecated
|====

[discrete]
[id="ocp-release-note-install-dep-rem_{context}"]
=== Installation deprecated and removed features

.Installation deprecated and removed tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.17 |4.18 |4.19

|`--cloud` parameter for `oc adm release extract`
|Deprecated
|Deprecated
|Deprecated

|CoreDNS wildcard queries for the `cluster.local` domain
|Deprecated
|Deprecated
|Deprecated

|`compute.platform.openstack.rootVolume.type` for {rh-openstack}
|Deprecated
|Deprecated
|Deprecated

|`controlPlane.platform.openstack.rootVolume.type` for {rh-openstack}
|Deprecated
|Deprecated
|Deprecated

|`ingressVIP` and `apiVIP` settings in the `install-config.yaml` file for installer-provisioned infrastructure clusters
|Deprecated
|Deprecated
|Deprecated

|Package-based {op-system-base} compute machines
|Deprecated
|Deprecated
|Deprecated

|Managing machines with the Cluster API for {azure-full}
|Not Available
|Technology Preview

> |Technology Preview

Reviewer 3: This somehow ended up in the wrong spot in 4.18. This goes in the Machine management TP table. I'll open a PR to fix 4.18.

Reviewer 3: PR with 4.18 fix: https://github.com/openshift/openshift-docs/pull/89397

Author: :heavy_check_mark:

Reviewer 3: thanks for merging that! :)

|`platform.aws.preserveBootstrapIgnition` parameter for {aws-first}
|Deprecated
|Deprecated
|Deprecated

|Installing a cluster on {aws-short} with compute nodes in {aws-short} Outposts
|Deprecated
|Deprecated
|Deprecated
|====

> [discrete]
> === Machine management deprecated and removed features
> 
> .Machine management deprecated and removed tracker
> [cols="4,1,1,1",options="header"]
> |====
> |Feature |4.17 |4.18 |4.19
> 
> |Managing machine with Machine API for {alibaba}
> |Removed
> |Removed
> |Removed
> 
> |Cloud controller manager for {alibaba}
> |Removed
> |Removed
> |Removed
> 
> |====

Reviewer 3: These can finally be removed since it's been 3 releases (4.16+)
```suggestion
////
[discrete]
=== Machine management deprecated and removed features

.Machine management deprecated and removed tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.17 |4.18 |4.19

|Placeholder
|Status
|Status
|Status

|====
////
```

.Edge computing Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.17 |4.18 |4.19

|Accelerated provisioning of {ztp}
|Technology Preview
|Technology Preview
|Technology Preview

|Enabling disk encryption with TPM and PCR protection
|Technology Preview
|Technology Preview
|Technology Preview

|====

[discrete]
[id="ocp-release-notes-installing-tech-preview_{context}"]
=== Installation Technology Preview features

.Installation Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.17 |4.18 |4.19

// All GA in 4.17 notes for oci-first
|Adding kernel modules to nodes with kvc
|Technology Preview
|Technology Preview
|Technology Preview

|Enabling NIC partitioning for SR-IOV devices
|General Availability
|General Availability
|General Availability

|User-defined labels and tags for {gcp-first}
|General Availability
|General Availability
|General Availability

|Installing a cluster on Alibaba Cloud by using Assisted Installer
|Technology Preview
|Technology Preview
|Technology Preview

|Mount shared entitlements in BuildConfigs in RHEL
|Technology Preview
|Technology Preview
|Technology Preview

> 
> |{product-title} on {oci-first}
> |General Availability
> |General Availability
> |General Availability

Reviewer 3: GA in 4.16
```suggestion
```

|Selectable Cluster Inventory
|Technology Preview
|Technology Preview
|Technology Preview

|Installing a cluster on {gcp-short} using the Cluster API implementation
|General Availability
|General Availability
|General Availability

> 
> |{product-title} on Oracle Compute Cloud@Customer (C3)
> |Not Available
> |General Availability
> |General Availability
> 
> |{product-title} on Oracle Private Cloud Appliance (PCA)
> |Not Available
> |General Availability
> |General Availability

Reviewer 3: These should be removed - they should have been feature blurbs in 4.18 (I think Brendan was going to update accordingly after 4.18 GA)
```suggestion
```

=== Machine Config Operator Technology Preview features

.Machine Config Operator Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.17 |4.18 |4.19

|Improved MCO state reporting (`oc get machineconfigpool`)
|Technology Preview
|Technology Preview
|Technology Preview

|On-cluster RHCOS image layering
|Technology Preview
|Technology Preview
|Technology Preview

|Node disruption policies
|General Availability
|General Availability
|General Availability

|Updating boot images for GCP clusters
|General Availability
|General Availability
|General Availability

|Updating boot images for AWS clusters
|Technology Preview
|General Availability
|General Availability
|====

[discrete]
[id="ocp-release-notes-machine-management-tech-preview_{context}"]
=== Machine management Technology Preview features

.Machine management Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.17 |4.18 |4.19

|Managing machines with the Cluster API for {aws-full}
|Technology Preview
|Technology Preview
|Technology Preview

|Managing machines with the Cluster API for {gcp-full}
|Technology Preview
|Technology Preview
|Technology Preview

> 

Reviewer 3: This goes here
```suggestion

|Managing machines with the Cluster API for {azure-full}
|Not Available
|Technology Preview
|Technology Preview

```

|Managing machines with the Cluster API for {vmw-full}
|Technology Preview
|Technology Preview
|Technology Preview

|Cloud controller manager for {ibm-power-server-name}
|Technology Preview
|Technology Preview
|Technology Preview

> 
> |Defining a {vmw-short} failure domain for a control plane machine set
> |General Availability
> |General Availability
> |General Availability

Reviewer 3: GA in 4.16
```suggestion
```

> 
> |Cloud controller manager for {alibaba}
> |Removed
> |Removed
> |Removed

Reviewer 3: Removed in 4.16
```suggestion
```

|====

[discrete]
[id="ocp-release-notes-monitoring-tech-preview_{context}"]
=== Monitoring Technology Preview features

.Monitoring Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.17 |4.18 |4.19

|Metrics Collection Profiles
|Technology Preview
|Technology Preview
|Technology Preview

|====

[discrete]
[id="ocp-release-notes-web-console-tech-preview_{context}"]
=== Web console Technology Preview features

.Web console Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.17 |4.18 |4.19

|{ols-official} in the {product-title} web console
|Technology Preview
|Technology Preview
|Technology Peview
|====

[discrete]
[id="ocp-release-notes-multi-arch-tech-preview_{context}"]
=== Multi-Architecture Technology Preview features

.Multi-Architecture Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.17 |4.18 |4.19

|`kdump` on `arm64` architecture
|Technology Preview
|Technology Preview
|Technology Preview

|`kdump` on `s390x` architecture
|Technology Preview
|Technology Preview
|Technology Preview

|`kdump` on `ppc64le` architecture
|Technology Preview
|Technology Preview
|Technology Preview

> 
> |Multiarch Tuning Operator
> |General Availability
> |General Availability
> |General Availability

Reviewer 3: GA in 4.16
```suggestion
```

|Technology Preview
|Technology Preview
|Technology Preview

|Updating the interface-specific safe sysctls list
|Technology Preview
|Technology Preview
|Technology Preview

|Egress service custom resource
|Technology Preview
|Technology Preview
|Technology Preview

|VRF specification in `BGPPeer` custom resource
|Technology Preview
|Technology Preview
|Technology Preview

|VRF specification in `NodeNetworkConfigurationPolicy` custom resource
|Technology Preview
|Technology Preview
|Technology Preview

|Host network settings for SR-IOV VFs
|General Availability
|General Availability
|General Availability

|Integration of MetalLB and FRR-K8s
|General Availability
|General Availability
|General Availability

|Automatic leap seconds handling for PTP grandmaster clocks
|General Availability
|General Availability
|General Availability

|PTP events REST API v2
|General Availability
|General Availability
|General Availability

|Customized `br-ex` bridge needed by OVN-Kuberenetes to use NMState
|Technology Preview
|General Availability
|General Availability

| Live migration to OVN-Kubernetes from OpenShift SDN
|General Availability
|Not Available

> |Not Available

Reviewer 3: This looks more like a removed feature than TP. Might be good to have someone in this area look at this one for placement

Author: Pinged Joe for clarity.

Reviewer 4: This is sort of a weird nwt thing. The TL;DR is this is correct. cc @stevsmit

> |User defined network segmentation

Reviewer 3: ```suggestion
|User-defined network segmentation
```

|{olmv1-first}
|Technology Preview
|General Availability
|General Availability

|{olmv1} runtime validation of container images using sigstore signatures
|Not Available
|Technology Preview
|Technology Preview
|====

[discrete]
[id="ocp-release-notes-operator-lifecycle-tech-preview_{context}"]
=== Operator lifecycle and development Technology Preview features

// "Operator lifecycle" refers to OLMv0 and "development" refers to Operator SDK

.Operator lifecycle and development Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.17 |4.18 |4.19

|{olmv1-first}
|Technology Preview
|General Availability
|General Availability

|Scaffolding tools for Hybrid Helm-based Operator projects
|Deprecated
|Removed
|Removed

|Scaffolding tools for Java-based Operator projects
|Deprecated
|Removed
|Removed
|====

[discrete]
[id="ocp-release-notes-rhcos-tech-preview_{context}"]
=== {rh-openstack-first} Technology Preview features

.{rh-openstack} Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.17 |4.18 |4.19

|{rh-openstack} integration into the {cluster-capi-operator}
|Technology Preview
|Technology Preview
|Technology Preview

> |Control Plane with `rootVolumes` and `etcd` on local disk

Reviewer 3: ```suggestion
|Control plane with `rootVolumes` and `etcd` on local disk
```

|Technology Preview

|Increasing the etcd database size
|Technology Preview
|Technology Preview
|Technology Preview

|Using {rh-rhacm} `PolicyGenerator` resources to manage {ztp} cluster policies
|Technology Preview
|Technology Preview
|Technology Preview

|Pinned Image Sets
|Technology Preview
|Technology Preview
|Technology Preview
|====

[discrete]
[id="ocp-release-notes-special-hardware-tech-preview_{context}"]
=== Specialized hardware and driver enablement Technology Preview features

.Specialized hardware and driver enablement Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.17 |4.18 |4.19
|====

[discrete]
[id="ocp-release-notes-storage-tech-preview_{context}"]
=== Storage Technology Preview features

.Storage Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.17 |4.18 |4.19

|AWS EFS storage CSI usage metrics
|General Availability
|General Availability
|General Availability

|Automatic device discovery and provisioning with Local Storage Operator
|Technology Preview
|Technology Preview
|Technology Preview

|Azure File CSI snapshot support
|Technology Preview
|Technology Preview
|Technology Preview

> 
> |Read Write Once Pod access mode
> |General Availability
> |General Availability
> |General Availability

Reviewer 3: GA in 4.16
```suggestion
```
